hack/test-cmd.sh

```diff
@@ -39,13 +40,22 @@ function start-proxy()
   stop-proxy
 
   kube::log::status "Starting kubectl proxy"
+
+  set-random-port
+
   # the --www and --www-prefix are just to make something definitely show up for
   # wait_for_url to see.
   kubectl proxy -p ${PROXY_PORT} --www=. --www-prefix=/healthz "$@" 1>&2 &
+
   PROXY_PID=$!
   kube::util::wait_for_url "http://127.0.0.1:${PROXY_PORT}/healthz" "kubectl proxy $@"
 }
 
+function set-random-port()
+{
+  PROXY_PORT=$(awk -v min=1 -v max=65535 'BEGIN{srand(); print int(min+rand()*(max-min+1))}')
```

Comment:
I wonder if there's an easy, portable way to tell if something is using a port. I'd suggest using netcat, but I'm not sure we can assume that's installed everywhere.